controller/array_action/array_mediator_svc.py

```diff
@@ -842,3 +871,192 @@ def get_array_fc_wwns(self, host_name):
             logger.error(msg="Failed to get array fc wwn. Reason "
                              "is: {0}".format(ex))
             raise ex
+
+    def _get_replication_endpoint_type(self, rcrelationship):
+        if self._identifier == rcrelationship.master_cluster_id:
+            return ENDPOINT_TYPE_MASTER
+        return ENDPOINT_TYPE_AUX
+
+    def _get_replication_other_endpoint_type(self, rcrelationship):
+        endpoint_type = self._get_replication_endpoint_type(rcrelationship)
+        if endpoint_type == ENDPOINT_TYPE_MASTER:
+            return ENDPOINT_TYPE_AUX
+        return ENDPOINT_TYPE_MASTER
+
+    @staticmethod
+    def _is_replication_idle(rcrelationship):
+        return rcrelationship.state == RCRELATIONSHIP_STATE_IDLE
+
+    @staticmethod
+    def _is_replication_disconnected(rcrelationship):
+        return 'disconnected' in rcrelationship.state
+
+    @staticmethod
+    def _is_replication_ready(rcrelationship):
+        return rcrelationship.state.startswith('consistent')
+
+    def _is_replication_endpoint_primary(self, rcrelationship, endpoint_type=None):
+        if not endpoint_type:
+            endpoint_type = self._get_replication_endpoint_type(rcrelationship)
+        return rcrelationship.primary == endpoint_type
+
+    @staticmethod
+    def _get_replication_copy_type(rcrelationship):
+        if rcrelationship.copy_type == 'global':
+            return config.REPLICATION_COPY_TYPE_ASYNC
+        return config.REPLICATION_COPY_TYPE_SYNC
+
+    def _generate_replication_response(self, rcrelationship, volume_internal_id, other_volume_internal_id):
+        copy_type = self._get_replication_copy_type(rcrelationship)
+        is_primary = self._is_replication_endpoint_primary(rcrelationship)
+        is_ready = self._is_replication_ready(rcrelationship)
+        return Replication(name=rcrelationship.name,
+                           volume_internal_id=volume_internal_id,
+                           other_volume_internal_id=other_volume_internal_id,
+                           copy_type=copy_type,
+                           is_primary=is_primary,
+                           is_ready=is_ready)
+
+    def _get_lsrcrelationship(self, filter_value):
```

Comment:
`_get_cli_rcrelationship`?
AFAIK `cli` letting us know that the object that the method returns is "storage object" and not the svc command in the name..

Reply:
since there's already a `_get_rcrelationship`, I think `_get_cli_rcrelationship` would be confusing.
how about just `_lsrcrelationship`?

Reply:
we have `get_volume` and `_get_cli_volume`..

Reply:
`get_volume` is public, and `cli` helps differentiate between a "server" volume and a "mediator" volume.
but in this case, `replication` is the "server" object and "rcrelationship" is the "mediator" object, so `cli` doesn't help in this case

Reply:
if we had `get_volume` and `_get_vdisk` for example, then we wouldn't need `cli` at all

Reply:
and how do we now that `vdisk` is a "mediator" or "server" object

Reply:
`vdisk` is an svc-only term